### ISSUE TITLE
PEP 585 runtime exceptions not recognized

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#1001]
## Environment data

-   Language Server version: 2021.2.4
-   OS and version: MacOS 10.15.7
-   Python version: **3.8.8**
-   `typeCheckingMode`: `strict`

## Expected behaviour

The following examples should be invalid without `from __future__ import annotations` in Python 3.7 and 3.8.

```py
import collections
import collections.abc
import contextlib
import re

var: collections.defaultdict[int, str]
var2: collections.abc.Iterable[int]  # and every other type from `collections.abc` (except `Hashable` and `Sized`)
var3: contextlib.AbstractContextManager[int]
var4: contextlib.AbstractAsyncContextManager[int]
var5: re.Pattern[str]
var6: re.Match[str]
```

https://docs.python.org/3/library/typing.html#corresponding-to-other-types-in-collections-abc

## Actual behaviour

Pylance doesn't infer them as errors.
Interestingly `collections.OrderedDict[int, str]`, `ChainMap`, `Counter` and `deque` are recognized correctly.

**Comment by erictraut** [comment]
Thanks for the suggestion.

I was able to address this for `collections.defaultdict`, but I can't do anything about the others because the stdlib type stubs simply alias the types in `typing.pyi`. From the perspective of Pyright, `collections.abc.Iterable` is simply an alias for `typing.Iterable`, so it can't differentiate between the two. If you would like to submit changes to typeshed's stubs so these become distinct types, I could easily add support.

The fix for `collections.defaultdict` will be in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.3.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202130-3-march-2021
